src/pgtz/src/timezone.rs

```diff
@@ -237,6 +237,11 @@ fn tokenize_timezone(value: &str) -> Result<Vec<TimeStrToken>, String> {
                 toks.push(TimeStrToken::TzName(substring.to_string()));
                 return Ok(toks);
             }
+            // PG allows arbitrary punctuation marks, which represent delimiters.
+            chr if chr.is_ascii_punctuation() => {
+                parse_num(&mut toks, &num_buf, split_nums, i)?;
```

Comment:
I don't know if I've ever looked at this code, will `parse_num` just ignore the punctuation?